multiapps-mta/src/test/java/org/cloudfoundry/multiapps/mta/resolvers/PropertiesResolverTest.java

```diff
@@ -13,41 +13,42 @@
-                         Arguments.of("moduleProperties.yaml", "test-list//foo/",
+                         Arguments.of("missing-index", "${test-list//foo/}",
                                       new Expectation(Expectation.Type.EXCEPTION, "Unable to resolve \"test-#test-list//foo/\"")),
-                         Arguments.of("moduleProperties.yaml", "tricky-map/just a key", new Expectation("foo")),
-                         Arguments.of("moduleProperties.yaml", "tricky-map/0/1", new Expectation("baz")),
-                         Arguments.of("moduleProperties.yaml", "tricky-map/one/two/", new Expectation("why")),
-                         Arguments.of("moduleProperties.yaml", "tricky-map/3/4/5", new Expectation("stop")));
+                         Arguments.of("corner-case-whitespace", "${tricky-map/just a key}", new Expectation("foo")),
+                         Arguments.of("corner-case-map-and-list", "${tricky-map/0/1}", new Expectation("baz")),
+                         Arguments.of("corner-case-slash-1", "${tricky-map/one/two/}", new Expectation("why")),
+                         Arguments.of("corner-case-slash-2", "${tricky-map/3/4/5}", new Expectation("stop")),
+                         Arguments.of("9-qux", "${NO_CIRCULAR_REF}", new Expectation("qux-qux-qux-qux-qux-qux-qux-qux-qux")),
+                         Arguments.of("9-qux-direct", "${foo}-${bar}-${baz}", new Expectation("qux-qux-qux-qux-qux-qux-qux-qux-qux")),
+                         Arguments.of("2-ha", "${NO_CIRCULAR_REF_IN_MAP/a}", new Expectation("ha-ha")),
+                         Arguments.of("circular-ref", "${CIRCULAR_REF}",
+                                      new Expectation(Expectation.Type.EXCEPTION, "Circular reference detected in \"test-#circular-ref\"")),
+                         Arguments.of("circular-ref-map", "${CIRCULAR_REF_MAP_IN_MAP/key/key}", new Expectation(Expectation.Type.EXCEPTION,
+                                                                                                                "Circular reference detected in \"test-#circular-ref-map\"")));
     }
 
     @ParameterizedTest
     @MethodSource
-    void testResolve(String modulePropertiesLocation, String parameterExpression, Expectation expectation) {
-        Map<String, Object> moduleProperties = TestUtil.getMap(modulePropertiesLocation, getClass());
-
-        tester.test(() -> testResolver.resolveReferenceInDepth(new Reference(null, parameterExpression), moduleProperties), expectation);
+    void testResolve(String parameterName, String parameterValue, Expectation expectation) {
+        PropertiesResolver testResolver = new PropertiesResolver(null, irrelevant -> replacementValues, ReferencePattern.PLACEHOLDER,
+                                                                 "test-", false, Collections.emptySet());
+        tester.test(() -> testResolver.visit(parameterName, parameterValue), expectation);
```

Comment:
Why invoked method was changed from `resolveReferenceInDepth` to `visit`?

Reply:
the previous method only tested a small portion of the code and couldn't resolve variables in depth (variables containing refs to other refs)